filebeat/harvester/log.go

```diff
@@ -87,6 +88,11 @@ func (h *Harvester) Harvest() {
 		if !h.sendEvent(event) {
 			return
 		}
+
+		if h.config.CloseTimeout > 0 && time.Since(startTime) > h.config.CloseTimeout {
+			logp.Info("Closing harvester because ttl was reached: %s", h.path)
+			return
+		}
```

Comment:
Alternative implementation could be to implement the timeout in a separate go routine the close `h.done`?